scim2-sdk-common/src/main/java/com/unboundid/scim2/common/messages/PatchOperation.java

```diff
@@ -153,11 +146,201 @@ public <T> List<T> getValues(final Class<T> cls)
     @Override
     public void apply(final ObjectNode node) throws ScimException
     {
-      JsonUtils.addValue(getPath() == null ? Path.root() :
-          getPath(), node, value);
+      Path path = (getPath() == null) ? Path.root() : getPath();
+      if (hasValueFilter(path))
+      {
+        validateAddOpWithFilter(path, value);
+        applyAddWithValueFilter(path, node, value);
+      }
+      else
+      {
+        JsonUtils.addValue(path, node, value);
```

Comment:
I tried really hard to make these changes follow the Visitor pattern in JsonUtils by adding a `CreatingNodeVisitor` type. However, I wasn't able to get this to work because the "traversal" nature of `traverseValues()` didn't play well with this. When someone typically wants to add a new value to an array, the value in the patch operation is set to an array of a single element. But patch requests with value selection filters don't set the value to an array. Hence, I had to resort to an if-statement to perform alternate logic for cases where a value selection filter is present.